files.go

```diff
@@ -248,6 +251,9 @@ func (api *Client) UploadFileContext(ctx context.Context, params FileUploadParam
 	} else if params.File != "" {
 		err = postLocalWithMultipartResponse(ctx, api.httpclient, "files.upload", params.File, "file", values, response, api)
 	} else if params.Reader != nil {
+		if params.Filename == "" {
```

Comment:
I'd just move this to the top and return it immediately. otherwise PR looks good.

Reply:
OK! I'll commit the changes.